File: Sort Colors.cpp
class Solution{
public:
    /**
     * @param nums: A list of integer which is 0, 1 or 2 
     * @return: nothing
     */    
    void sortColors(vector<int> &nums) {
        // write your code here
        int size = nums.size();
        int left = 0, right = size - 1;
        for (int i = 0; i <= right;)
        {
            // i = max(left, i);
            if (nums[i] == 0)
            {
                swap(nums[i], nums[left]);
                ++left;
                ++i;
            }
            else if (nums[i] == 2)
            {
                swap(nums[i], nums[right]);
                --right;
            }
            else if (nums[i] == 1)
            {
                ++i;
            }
        }
    }
};

File: Sort Colors.py
class Solution:
    """
    @param nums: A list of integer which is 0, 1 or 2 
    @return: nothing
    """
    def sortColors(self, nums):
        # write your code here
        
        if not nums:
            return None
        length = len(nums)
        left, pivot, right = 0, 0, length - 1
        while left <= right:
            if nums[left] == 0:
                nums[pivot], nums[left] = nums[left], nums[pivot]
                pivot += 1
                left += 1
            elif nums[left] == 1:
                left += 1
            else:
                nums[left], nums[right] = nums[right], nums[left]
                right -= 1
        